AbigailDawson/knownative
# KN-8: Refactor File Structure

### Describe your changes:

This PR restructures the KnowNative files into a compartmentalized format. See [SlackThread](https://knownative.slack.com/archives/C06QLTVBSEB/p1727144743400009?thread_ts=1726618476.764239&cid=C06QLTVBSEB) for a more in depth written description.

I made a diagram to demonstrate the general idea of the file structure. Each Box represents a folder. You can see how things get nested by what they are related to. A good way to look at this is what is being imported on each file and organize that way. If you notice that something is being reused elsewhere, that is a good indication that it could be refactored and moved into either a higher level, or into the ui-components (see difficultytag, which is used both in the LeftNav and in the MainContent and thus lives in the components folder of the demo page)

![Screenshot 2024-10-01 at 5 46 51 PM](https://github.com/user-attachments/assets/c08f5160-92d6-42a8-8bc8-c6f174d9d082)

### Issue ticket number and link:

https://huly.app/workbench/knownative/tracker/KN-8

### Notes:

None

### How to test:

Click through app and its functions to verify nothing broke or is missing

<sub><a href="https://huly.app/guest/knownative?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NmZjOThjZjQzYjBiYWU0OTVhNTUwNDMiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWJpZ2FpbGRhd3NvLWtub3duYXRpdmUtNjYzMjgxYjEtODdiZTE2ZDY5OS0yYTZiY2QifQ.XetsuePZz_lcDECG8_MwuJvVw6e0Sp07GW_QKpAVgMA">Huly&reg;: <b>GITHB-133</b></a></sub>
